### DISCUSSION TITLE
Should "Actions expect form-encoded data" not return http status code 400?

### DISCUSSION THREAD

**Discussion by alexbjorlig** [sveltejs/kit#11248] [Q&A]
If you POST JSON to a form action, sveltekit will throw `Actions expect form-encoded data (received application/json)`.

This will then bubble to `HandleServerError`, and result in an Error with HTTP status code 500.

I think it would be better if the error was either [415](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/415) or [400](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400).

One of the issues with the current implementation is that if you report to Sentry in the error [handler](https://kit.svelte.dev/docs/hooks#shared-hooks-handleerror), you will report an error when people submit wrong data (something that will create a lot of noise in Sentry). Better to reserve HTTP 500 for situations where we don't have better.

I might be missing something. If not, then I would be happy to submit a PR.

**Comment by teemingc** [comment]
Hi Alex. That sounds like a great idea. I think the http status code 415 is fitting. I've created an issue for this from this discussion https://github.com/sveltejs/kit/issues/11251

**Reply by alexbjorlig** [reply]
Ok great, will try to create my first PR for kit then 😅

**Reply by alexbjorlig** [reply]
ok, submitted [here](https://github.com/sveltejs/kit/pull/11255) 🚀
